rrs/views.py

```diff
@@ -0,0 +1,1005 @@
+        if not milestone:
+            raise Exception('No milestones defined for release %s' % release.name)
+        return reverse('rrs_recipes', args=(maintplan.name, release.name, milestone.name))
+
+class MaintenancePlanRedirect(RedirectView):
+    permanent = False
+
+    def get_redirect_url(self, maintplan_name):
+        maintplan = get_object_or_404(MaintenancePlan, name=maintplan_name)
+        release = Release.get_current(maintplan)
+        if not release:
+            raise Exception('No releases defined for maintenance plan %s' % maintplan.name)
+        milestone = Milestone.get_current(release)
+        if not milestone:
+            raise Exception('No milestones defined for release %s' % release.name)
+        return reverse('rrs_recipes', args=(maintplan.name, release.name, milestone.name))
+
+
+def _check_url_params(upstream_status, maintainer_name):
+    get_object_or_404(Maintainer, name=maintainer_name)
+
+    found = 0
+    for us in RecipeUpstream.RECIPE_UPSTREAM_STATUS_CHOICES_DICT.keys():
+        if us == 'D': # Downgrade is displayed as Unknown
+            continue
+
+        if RecipeUpstream.RECIPE_UPSTREAM_STATUS_CHOICES_DICT[us] == upstream_status:
+            found = 1
+            break
+
+    if found == 0:
+        raise Http404
+
+def _get_layer_branch_url(branch, layer_name):
+    return reverse_lazy('layer_item', args=(branch, layer_name))
+
+class Raw():
+    """ Raw SQL call to improve performance
+
+        Table abbrevations:
+        re:     Recipe
+        ma:     Maintainer
+        reup:   Recipe Upstream
+        reupg:  Recipe Ugrade
+        rema:   Recipe Maintainer
+        remahi: Recipe Maintainer History
+    """
+
+    @staticmethod
+    def get_re_by_mantainer_and_date(maintainer, date_id):
+        """ Get Recipes based on Maintainer and Recipe Maintainer History """
+        recipes = []
+        cur = connection.cursor()
+
+        cur.execute("""SELECT DISTINCT rema.recipesymbol_id
+                        FROM rrs_recipemaintainer AS rema
+                        INNER JOIN rrs_maintainer AS ma
+                        ON rema.maintainer_id = ma.id
+                        WHERE rema.history_id = %s 
+                        AND ma.name = %s;
+                    """, [date_id, maintainer])
+
+        for re in cur.fetchall():
+            recipes.append(re[0])
+        return recipes
+
+    @staticmethod
+    def get_ma_by_recipes_and_date(recipes_id, date_id=None):
+        """ Get Maintainer based on Recipes and Recipe Upstream History """
+        stats = []
+
+        if date_id:
+            qry = """SELECT rema.recipesymbol_id, ma.name
+                    FROM rrs_recipemaintainer AS rema
+                    INNER JOIN rrs_maintainer AS ma
+                    ON rema.maintainer_id = ma.id
+                    WHERE rema.history_id = %s
+                    AND rema.recipesymbol_id IN %s;"""
+            cur = connection.cursor()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the following resource: *cur*. The resource is allocated by call *connection.cursor*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close *cur* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).